### PR TITLE
MultiRule should pass correctly the BindingContext

### LINKED ISSUE #4058: BooleanPropertyNaming not reporting any violation
## Expected Behavior
`BooleanPropertyNaming` was added in 1.18.0. After enabling this in `detekt.yml`, the violations should be reported.

## Observed Behavior
Not even a single violation is reported.

## Steps to Reproduce
I have a multi-module project. I have even setup custom Detekt rules in my project.
Here is the root level build.gradle.
```
plugins {
    id "io.gitlab.arturbosch.detekt" version "1.18.0"
}

apply plugin: 'io.gitlab.arturbosch.detekt'
dependencies {
    detektPlugins project(':static-analysis:detekt-rules') //custom Detekt rules
}

tasks.register("detektAll", Detekt) { //custom Gradle task
    description = "Custom Detekt build for all modules"
    parallel = true
    setSource(file(projectDir))
    config.setFrom(files("$rootDir/detekt.yml"))
    jvmTarget = "1.8"
    classpath.setFrom(project.configurations.getByName("detekt"))
    include("**/*.kt")
    exclude("**/build/**")
    reports {
        html {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.html")
        }
        xml {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.xml")
        }
        txt.enabled = false
    }
}
```

Part of `detekt.yml`
```
naming:
  active: true
  severity: error
  BooleanPropertyNaming:
    active: true
  ClassNaming:
    active: true
    classPattern: '[A-Z][a-zA-Z0-9]*'
  ForbiddenClassName:
    active: false
    forbiddenName: [Manager]
  MemberNameEqualsClassName:
    active: true
    ignoreOverridden: true
  NoNameShadowing:
    active: true
  NonBooleanPropertyPrefixedWithIs:
    active: true
```
    
When I copy the code for `BooleanPropertyNaming` in my custom rules project `static-analysis:detekt-rules`, the violations are reported.

## Context
Just trying to enable `BooleanPropertyNaming` on my project.

## Your Environment
* Version of detekt used: 1.18.0
* Version of Gradle used (if applicable): 7.0.0
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS 10.15.6
* Link to your project (if it's a public repository):

### PR DESCRIPTION
**Pull request by cortinico**
Fixes #4058

The rule `BooleanPropertyNaming` that was recently added relies on Type Resolution. That rule was plugged inside a `MultiRule` instead of a classical `RuleSetProvider`.

Given that rules rely on inheritance to access a `BindingContext`, the `MultiRule` should pass over the `BindingContext` to the rules he's managing.

The result was that `BooleanPropertyNaming` was effectively not running at all. This PR is fixing it.

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt [hunk 1]
```diff
@@ -168,7 +168,7 @@ class UseIsNullOrEmpty(config: Config = Config.empty) : Rule(config) {
 
         private val stringClass = StandardNames.FqNames.string.toSafe()
 
-        private val isEmptyFunctions = collectionClasses.map { FqName("$it.isEmpty") } +
+        private val emptyCheckFunctions = collectionClasses.map { FqName("$it.isEmpty") } +
```
Comment (chao2zhang) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt:171]: Is this a renaming that's not relevant? Just checking if we are accidentally cherry-picking other commits.

**Review by chao2zhang — APPROVED** [review]
Good find!!!

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt (hunk 1 above)
Comment (cortinico) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt:171]: It's relevant actually. `BooleanPropertyNaming` was not the only rule affected by this. The `NonBooleanPropertyPrefixedWithIs` was affected as well and now got triggered on the codebase.

**Review by schalkms — APPROVED** [review]

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt (hunk 1 above)
Comment (chao2zhang) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseIsNullOrEmpty.kt:171]: It is now catching issues. Got it!

**Pull request merged**